src/binder/clrprivbinderassemblyloadcontext.cpp

```diff
@@ -97,16 +113,19 @@ HRESULT CLRPrivBinderAssemblyLoadContext::BindAssemblyByName(IAssemblyName     *
                 }
             }
         }
-    }
 
-    IF_FAIL_GO(hr);
+        IF_FAIL_GO(hr);
+
+        // Extract the assembly reference.
+        //
+        // For TPA assemblies that were bound, TPABinder
+        // would have already set the binder reference for the assembly, so we just need to
+        // extract the reference now.
+        *ppAssembly = pCoreCLRFoundAssembly.Extract();
+
+        tracer.SetResult(hr, pCoreCLRFoundAssembly);
```

Comment:
This is going to result in an event where the result is attributed to the last stage from this `ResolutionAttemptedOperation` even though it could be from one of the stages in `BindUsingHostAssemblyResolver`?

Reply:
Changed how the class gathers information to fire events.   Should work even when exceptions are thrown.